sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -389,23 +389,13 @@ public Mono<PathInfo> upload(BinaryData data, ParallelTransferOptions parallelTr
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<PathInfo> upload(Flux<ByteBuffer> data, ParallelTransferOptions parallelTransferOptions,
         boolean overwrite) {
+        DataLakeRequestConditions requestConditions = new DataLakeRequestConditions();
 
-        Mono<Void> overwriteCheck;
-        DataLakeRequestConditions requestConditions;
-
-        if (overwrite) {
-            overwriteCheck = Mono.empty();
-            requestConditions = null;
-        } else {
-            overwriteCheck = exists().flatMap(exists -> exists
-                ? monoError(LOGGER, new IllegalArgumentException(Constants.BLOB_ALREADY_EXISTS))
-                : Mono.empty());
-            requestConditions = new DataLakeRequestConditions()
-                .setIfNoneMatch(Constants.HeaderConstants.ETAG_WILDCARD);
+        if (!overwrite) {
```

Comment:
Looking at the append and flush used by Datalake it's the same concept as block blob where append doesn't take an ETag. So this has the same issue as my previous comment

Reply:
So in this case, is it better to leave the overwrite logic as it was being done previously?

CC @rickle-msft